Source/Core/DolphinQt/DiscordJoinRequestDialog.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#ifdef USE_DISCORD_PRESENCE
+
+#include <QGridLayout>
+#include <QLabel>
+#include <QPixmap>
+#include <QPushButton>
+
+#include <discord-rpc/include/discord_rpc.h>
+
+#include "Common/HttpRequest.h"
+#include "Common/StringUtil.h"
+
+#include "DolphinQt/DiscordJoinRequestDialog.h"
+
+DiscordJoinRequestDialog::DiscordJoinRequestDialog(QWidget* parent, const char* id,
+                                                   const std::string& discord_tag,
+                                                   std::string avatar)
+    : QDialog(parent), m_user_id(id), m_close_timestamp(std::time(nullptr) + s_max_lifetime_seconds)
+{
+  setWindowTitle(tr("Request to Join Your Party"));
+  setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+
+  QPixmap avatar_pixmap;
+
+  if (avatar.empty())
+  {
+    const std::string avatar_endpoint =
+        StringFromFormat("https://cdn.discordapp.com/avatars/%s/%s.png", id, avatar.c_str());
+
+    Common::HttpRequest request;
+    Common::HttpRequest::Response response = request.Get(avatar_endpoint);
+
+    if (response.has_value())
+      avatar_pixmap.loadFromData(response->data(), static_cast<uint>(response->size()), "png");
+  }
+
+  CreateLayout(discord_tag, avatar_pixmap);
+  ConnectWidgets();
+}
+
+std::time_t DiscordJoinRequestDialog::GetCloseTimestamp() const
+{
+  return m_close_timestamp;
+}
+
+void DiscordJoinRequestDialog::CreateLayout(const std::string& discord_tag, const QPixmap& avatar)
+{
+  m_main_layout = new QGridLayout;
+
+  m_invite_button = new QPushButton(QString::fromWCharArray(L"\u2714 Invite"));
```

Comment:
Any reason why this string (+ line below) is not marked as translatable?

Reply:
Sorry, I wasn't aware of what that function actually did. I thought it was short for ``QString::fromUtf8``, as I judged it by it's behavior at first glance. I should have looked at the qt docs to make sure.